src/main/java/org/springframework/data/couchbase/SimpleCouchbaseClientFactory.java

```diff
@@ -74,7 +76,7 @@ private SimpleCouchbaseClientFactory(final Supplier<Cluster> cluster, final Stri
 
 	@Override
 	public CouchbaseClientFactory withScope(final String scopeName) {
-		return new SimpleCouchbaseClientFactory(cluster, bucket.name(), scopeName);
+		return new SimpleCouchbaseClientFactory(cluster, bucket.name(), scopeName != null ? scopeName : getScope().name());
```

Comment:
if this is null, should this use the default scope? Or should we add a withDefaultScope?

Reply:
getScope() is the scope of the clientFactory - that comes from AbstractCouchbaseConfiguration.getScope() or an overriding method.